src/Browser/Avalonia.Browser/BrowserAppBuilder.cs

```diff
@@ -78,8 +87,35 @@ public static class BrowserAppBuilder
             .AfterApplicationSetup(_ =>
             {
                 lifetime.View = new AvaloniaView(mainDivId);
-            })
-            .SetupWithLifetime(lifetime);
+            });
+
+        if (BrowserWindowingPlatform.IsManagedDispatcherEnabled)
```

Comment:
I suspect that  `AvaloniaLocator.Current.GetService<BrowserPlatformOptions>()?.PreferManagedThreadDispatcher != false;` check will always succeed regardless of the `options` parameter passed to StartBrowserAppAsync. Only one registered with `.With` will be used

Reply:
We already have `AvaloniaLocator.CurrentMutable.Bind<BrowserPlatformOptions>().ToConstant(options);` for that case. It's auto-registered.